packages/mobile/src/send/utils.ts

```diff
@@ -0,0 +1,288 @@
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { FeeType } from 'src/fees/actions'
+import { E164NumberToAddressType, RecipientVerificationStatus } from 'src/identity/reducer'
+import { Recipient, RecipientKind } from 'src/recipients/recipient'
+import { ConfirmationInput, SecureSendPhoneNumberMapping, TransactionData } from 'src/send/reducers'
+import Logger from 'src/utils/Logger'
+
+export const formatDisplayName = (displayName: string) => {
+  // Default displayName is "Mobile #" when no contact is saved
+  // To make the copy nicer, we make the displayName "your contact" isntead
+  if (displayName === 'Mobile #') {
+    return { displayName: 'your contact', displayNameCapitalized: 'Your contract' }
+  }
+
+  return { displayName, displayNameCapitalized: displayName }
+}
+
+export const getAddressFromPhoneNumber = (
```

Comment:
Please move these back. They aren't utils for the send flow. They're utils for access identity related data, which are used by places in code other than the send flow

Reply:
I am open to moving this wherever, but think it's worth pointing out that `getAddressFromPhoneNumber` is only used in `getConfirmationInput`, which is only used in `SendConfirmation.tsx` and `PaymentRequestConfirmation.tsx`

More broadly, `getAddressFromPhoneNumber` as it is designed cannot be used outside of the context of preparing to send a transaction. Attempting to use it outside of this context  risks it breaking because we have not yet gathered the data required to disambiguate an phone number that maps to multiple addresses, which is where Secure Send comes in.

Reply:
Valid points, thanks! Not sure why I thought there may be other consumers of this atm. 

My concern is mainly to prevent identity related complexity from leaking to multiple places in the app. Keeping it confined to the `identity` folder seems more sustainable to me. Also I find it's bets to avoid putting meaningful logic into `utils` files which become a dumping ground over time.

What's the issue with having it be in `identity` even if send flow is the only consumer for now? Just an idea but we could have this structured as a saga that navigates to the Secure Send screen as necessary, like what I've done for phone privacy here: https://github.com/celo-org/celo-monorepo/pull/3777/files#diff-2f75b310de275b594e6658f9d0eb23bfR215

Reply:
Cool that all makes sense to me. I see your point about grouping similar logic and agree. No issue with moving it (will do that), just wanted to make sure we were on the same page about how it was being utilized.

Pairing the address fetching and the navigation together into one saga was something I considered and actually liked more from a technical perspective as the logic is more contained. The hangup for me was that there would be a 1-2 second lag (due to the attestation lookup) between the user action and the eventual navigation. Did the PGPNP implementation have a similar trade off? I am very open to revisiting this!

Reply:
I don't think PGPNP had that lag. Can you clarify what caused that 1-2 second lag and why it doesn't happen with the current implementation? Def not saying you need to change it btw, just so I understand.

Reply:
I believe the lag is just due to the async nature of the call we are making. In the current implementation, we are making this call on `componentDidMount` when the user hits the `SendAmount` screen. That way, by the time they press "send", we already know which direction we need to route them so they'll never really notice any type of loading time.